src/main/scala/Hierarchy.scala

```diff
@@ -0,0 +1,125 @@
+package d3v4
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.{JSExportAll, JSImport}
+
+/**
+ * @see [[https://github.com/d3/d3-hierarchy]]
+ * @since January, 2021
+ * @author <a href="mailto:michael@ahlers.consulting">Michael Ahlers</a>
+ */
+@JSImport("d3-hierarchy", JSImport.Namespace)
+@js.native
+object d3hierarchy extends js.Object {
+
+  /** @see [[https://github.com/d3/d3-hierarchy#hierarchy]] */
+  def hierarchy[Datum](data: Datum): Hierarchy[Datum] = js.native
+
+  /** @see [[https://github.com/d3/d3-hierarchy#hierarchy]] */
+  def hierarchy[Datum](data: Datum, children: js.Function1[Datum, js.Array[Datum]]): Hierarchy[Datum] = js.native
+
+  /** @see [[https://github.com/d3/d3-hierarchy#pack]] */
+  def pack[Datum](): Pack[Datum] = js.native
+
+  /** @see [[hierarchy]] */
+  @js.native
+  trait Hierarchy[Datum] extends js.Object {
+
+    def data: Datum = js.native
+    def depth: js.UndefOr[Int] = js.native
+    def height: js.UndefOr[Int] = js.native
+    def parent: js.UndefOr[this.type] = js.native
+    def children: js.UndefOr[js.Array[this.type]] = js.native
```

Comment:
Not part of any external type since they're managed entirely by hierarchy. I was wondering (and attempted) to make this similar to `SimulationNode` but that approach didn't pan out (since the types don't match callbacks).

Reply:
If it works well for you, that's fine.